cmd/serve/serve.go

```diff
@@ -56,6 +56,8 @@ func ServeCmd() *cobra.Command {
 			api := api.New(config.API.Size, metrics)
 			aio := aio.New(config.AIO.Size, metrics)
 
+			config.API.Subsystems.Http.Auth = viper.Get("auth").([]interface{})
```

Comment:
We can remove this

Reply:
Yup, removed it . Thanks for being with me @dfarr .

Now , i have a doubt , as am still learning, I saw that in ``` cmd/serve/serve.go``` we call ```config, err := util.NewConfig()``` which basically calls ```viper.Unmarshal(&config)``` on the config file and will fill up our Config struct and all its fields and sub-fields. Essentially picking up the creds present there.

Now, either the user provides additional creds on commandLine too sing the flag we created or there are no creds in config but are only provided via the command line; in that case how would our ```Config.Auth``` get filled ? Would we need a function to be created which looks for a ```api-http-auth``` flag being invoked and accordigly initialises ```config.Auth``` with the values recieved from flag ?

Reply:
Good question! Command line arguments take precedent over the config file, which is what we want. You can test all this out locally by running serve with differing command line arguments/config file values.
```
go run ./... serve
```

To interact with the server you can use the cli
```
go run ./... promises get foo
go run ./... promises get foo --username foo --password foo
```

Reply:
I ran it and loved to see how things were working seamlessly. Passed dummy creds to server , then Created a dummy promise ID and made these ```get``` requests by providing credentials on command line. Loved seeing the appropriate responses.

Learned the proper flow and usage of cobra commands and the logic of binding flags to viper .


Thanks for the guidance @dfarr